client/src/pages/Directory.tsx

```diff
@@ -0,0 +1,601 @@
+import React, {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useState,
+} from 'react';
+import { Dropdown, Button, Input } from 'semantic-ui-react';
+
+import { IUser } from '../../../common/index';
+import Sidebar from '../components/Sidebar';
+import UserModal from '../components/UserModal/UserModal';
+import { getUsers, isError } from '../utils/apiWrapper';
+import { pages } from '../utils/enums';
+import { parseArrayToSemanticDropdownOptions } from '../utils/helpers';
+import SSW from '../assets/ssw-form-header.png';
+import DefaultProfile from '../assets/default_profile.png';
+import '../css/Directory.css';
+
+interface IFilterKeys {
+  role: string;
+  date: string;
+  interests: string[];
+  teams: string[];
+}
+
+interface ISearchState {
+  users: IUser[];
+  query: string;
+  isLoading: boolean;
+}
+
+interface ISearchAction {
+  type: SearchAction;
+  query: string;
+  users: IUser[];
+}
+
+interface IModalState {
+  isOpen: boolean;
+  user?: IUser;
+}
+
+interface IModalAction {
+  type: ModalAction;
+  isOpen: boolean;
+  user?: IUser;
+}
+
+const initialFilterKeys: IFilterKeys = {
+  role: '',
+  date: '',
+  interests: [],
+  teams: [],
+};
+
+enum SearchAction {
+  CLEAN_QUERY = 'CLEAN_QUERY',
+  INITIALIZE_USERS = 'INITIALIZE_USERS',
+  START_SEARCH = 'START_SEARCH',
+  UPDATE_QUERY = 'UPDATE_QUERY',
+  FINISH_SEARCH = 'FINISH_SEARCH',
+}
+
+const initialSearchState: ISearchState = {
+  users: [],
+  query: '',
+  isLoading: false,
+};
+
+/**
+ * Reducer for the search state.
```

Comment:
I used a reducer to prevent having a lot a lot of state variables. So rather than having like 15 state variables, the reducers help tether the states that behave very closely together into one state that is easier to manage